// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.dsc";
option java_outer_classname = "DSC";
option objc_class_prefix = "DSC";

package dsc;

service Dsc {
  // rpc Add (addMsg) returns (addReply) {}

  // rpc ImageHW (imageMsg) returns (imageReply) {}

  rpc Humidity (humidityMsg) returns (humidityReply) {}

  rpc Temp (tempMsg) returns (tempReply) {}

  rpc Pressure (pressureMsg) returns (pressureReply) {}
}

message humidityMsg {
  int32 h = 1;
}

message tempMsg {
  int32 t = 1;
}

message pressureMsg {
  int32 p = 1;
}

message humidityReply {
  // %age relative humidity
  float humidity = 1;
}

message tempReply {
  // celsius
  float temp = 1;
}

message pressureReply {
  // Millibars
  float pressure = 1;
}

/*
message addMsg {
  int32 a = 1;
  int32 b = 2;
}

message addReply {
  int32 addsum = 1;
}

message imageMsg {
  bytes img = 1;
}

message imageReply {
  int32 w = 1;
  int32 h = 2;
}
*/